04.ls/ls_command.rb

```diff
@@ -5,47 +5,52 @@
 COLUMN_COUNT = 3
 
 def main
-  files = if a_option?
-            fetch_files(option: 'a')
+  options = load_options
+  files = fetch_files
+
+  files = if options.key?(:a)
+            files
+          else
+            files.filter { |file| file unless file.start_with?('.') }
+          end
+
+  files = if options.key?(:r)
+            files.sort.reverse
           else
-            fetch_files
+            files.sort
           end
+
   puts format_to_show(files)
 end
 
-def a_option?
-  has_a = false
+def load_options
+  options = {}
 
   opt = OptionParser.new
-  opt.on('-a') { |v| has_a = v }
+  opt.on('-a') { |v| options[:a] = v }
+  opt.on('-r') { |v| options[:r] = v }
   opt.parse!(ARGV)
 
-  has_a
+  options
 end
 
-def fetch_files(option: '')
-  case option
-  when ''
-    Dir.foreach(dir_path).filter_map { |file| file unless file.start_with?('.') }
-  when 'a'
-    Dir.foreach(dir_path).map { |file| file }
-  end
+def fetch_files
+  Dir.foreach(dir_path).map { |file| file }
```

Comment:
```suggestion
  Dir.foreach(dir_path).to_a
```

もしくはentriesメソッドを使うとか？

https://docs.ruby-lang.org/ja/latest/method/Dir/s/entries.html

だけで済みそうです。